multimedia/welle.io/Portfile

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               qt5 1.0
+PortGroup               github 1.0
+PortGroup               cmake 1.1
+
+#PortGroup               compilers 1.0
+
+github.setup            AlbrechtL welle.io 2.0-beta3 v
+github.tarball_from     archive
+name                    welle.io
+categories              multimedia
+maintainers             nomaintainer
+
+description             DAB/DAB+ Software Radio
+long_description        This is an open source DAB and DAB+ software defined radio (SDR)  \
+                        with support for rtl-sdr (RTL2832U) and airspy. It supports  \
+                        high DPI and touch displays and it runs even on cheap computers  \
+                        like Raspberry Pi 2/3 and 100€ China Windows 10 tablets.
+
+platforms               darwin
+license                 GPL-2
+license_noconflict      openssl
+
+homepage                https://www.welle.io/
+
+checksums               rmd160  05113c73dc7b97fed256d91cd6a971dabc30c1ce \
+                        sha256  99acaea8386b092ebc80cd6c689ab0106a9f74ad121c04d0ae893f4e35a5a3e0 \
+                        size    1613717
+
+qt5.depends_component   qtbase \
+                        qtcharts \
+                        qtdeclarative \
+                        qtmultimedia \
+                        qtquickcontrols \
+                        qtquickcontrols2
+
+depends_lib-append      port:faad2 \
+                        port:fftw-3-single \
+                        port:lame \
+                        port:libusb \
+                        port:mpg123
+
+default_variants        +rtlsdr +soapysdr +airspy
+
+variant rtlsdr description {Add RTL-SDR support} {
+    configure.pre_args-append \
+        -DRTLSDR=ON
+    depends_lib-append port:rtl-sdr
+}
+
+variant soapysdr description {Add SoapySDR support} {
+    configure.pre_args-append \
+        -DSOAPYSDR=ON
+    depends_lib-append port:SoapySDR
+}
+
+variant airspy description {Add Airspy support} {
+    configure.pre_args-append \
+        -DAIRSPY=ON
+    depends_lib-append port:airspy
+}
+
+variant morepatches description {Additional patches with bugfixes and features} {
```

Comment:
I don't understand the point of making this a variant. Why not always apply these patches if they are bug fixes and add features?

Reply:
To easily be able to determine whether a bug is comming from these various patches or from upstream. Because enabling them by default would mean having to make support for them specifically for this port.

Reply:
I'm not sure I understand what you mean by "make support". We support all ports including their variants. In other words, being in a variant doesn't change our support stance for this port. No other port does something like this so I'm not convinced that patches should be applied that may or may not work. Can you just apply the ones are known to work? Honestly, we are just porting the upstream software. So any patches should be submitted to the upstream project instead of in Macports. I think this variant and the patches should be removed entirely.

Reply:
Theses patches do all work, so there shouldn't be any problem applying them in all cases. However this is a slightly diverging and anticipating what will come in the next releases of welle.io
Some of them have already been accepted upstream and will go into 2.0 final, others will wait version 2.1.

What I can do is :
- apply only those accepted upstream (but not yet released) & remove those that will go into 2.1
- don't apply any patch to be fully in line with upstream (2.0beta3)

Let's see what the others say and I will just do the way you recommend.

Reply:
I would be OK with always (i.e. not in a variant) including the patches that are accepted upstream for the 2.0 release.

Reply:
Yah, I'm not sure about this idea that having a "debugging our patches" variant seems like a good idea. If you don't trust the patches, don't include them, if you do trust them, do. If someone complains about a bug, the maintainer gets to figure out what the problem is.

Reply:
Ok, I removed the morepatches variant and related patch files and instead of using 2.0beta3+"patches accepted upstream" I just took the last git commit.

Reply:
I agree with removing the `morepatches` variant for the reasons discussed. Thanks for doing it.

Reply:
Thank you!